### ISSUE TITLE
Cannot initialize device-mapper. Is dm_mod kernel module loaded?

### ISSUE THREAD

**Issue by dannyboyw** [Entware-ng/Entware-ng#43]
I am not sure if this is a bug, but I am encountering an issue while trying to mount a LUKS partition with command:
cryptsetup luksOpen /dev/sda1 mydmname
using the latest (3.0.0.4.378_9313) stock AsusWRT on my RT-AC3200.  

I followed the “Install on Asus stock firmware” instructions and was able to 
opkg install update
opkg install cryptsetup nano

Nano works as expected.  Am I missing other deps.  I am assuming cryptsetup would load dm_mod on call.  
modprobe -l | grep dm 
returns unrelated module cdc-wdm.ko

Thank you for your help and hard work!
Dan

**Comment by ryzhovau** [comment]
Entware-ng is firmware independent repo, no any kernel modules can be provided.

Try to find modules archive in the place where you get current firmware. Both kernel and modules must be from the same build, or you get bunch of strange errors otherwise.

**Comment by ryzhovau** [comment]
> latest (3.0.0.4.378_9313) stock AsusWRT on my RT-AC3200.

Well, it's time to download last GPL archive, change kernel config to compile `dm-mod.ko`, `aes.ko`, `sha256.ko` and `dm-crypt.ko`. This modules should be loaded before running `cryptsetup`.

I tested LUKS before on RT-N66U ([thread in Russian](http://wl500g.info/showthread.php?30367-cryptsetup-luks&p=254114#post254114)), it was painfully slow:)

**Comment by dannyboyw** [comment]
Thank you very much!  I came across [this] (https://vip.asus.com/forum/view.aspx?id=20121022225145408&board_id=11&model=RT-N66U%20%28VER.B1%29&page=1&SLanguage=en-us)  as well which inspired me to try on beefier hardware.  I am hoping with USB 3.0 drive and dual cores that it will be reasonable.  My policy is that all personal data is encrypted, no exceptions.  If the performance is not reasonable, I will have to go back to dedicated box, which has slim specs (Pentium 3 single core 1.8gHz/256MB) and I was able to stream 1080p video from encrypted volume over LAN without issue (I never measured actual throughput, but I will if I get this working).  Thanks again for your help!
